src/modules/tenant/service.js

```diff
@@ -63,6 +64,7 @@ export default class TenantService extends Service {
 			type: types.TENANTS_CHANGED,
 			tenants_list,
 			current: current_tenant,
+			invalid: invalid
```

Comment:
No, that's unnecessary and incorrect state.
Don't emit "tenant changes" when the authorisation fails - from the code, it seems to me that you don't need to have it at all.